src/wayland/seat/pointer.rs

```diff
@@ -305,9 +305,21 @@ where
             }
             // axis
             if details.axis.0 != 0.0 {
+                if ptr.version() >= 9 {
+                    ptr.axis_relative_direction(
+                        WlAxis::HorizontalScroll,
+                        details.relative_direction.0.into(),
+                    );
+                }
                 ptr.axis(details.time, WlAxis::HorizontalScroll, details.axis.0);
             }
             if details.axis.1 != 0.0 {
+                if ptr.version() >= 9 {
+                    ptr.axis_relative_direction(
+                        WlAxis::HorizontalScroll,
```

Comment:
```suggestion
                        WlAxis::VerticalScroll,
```
well well well if it isn't the copy-paste mistake again